src/Mvc/Mvc.ViewFeatures/src/Rendering/SystemTextJsonHelper.cs

```diff
@@ -0,0 +1,29 @@
+﻿
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Html;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.Rendering
+{
+    internal class SystemTextJsonHelper : IJsonHelper
+    {
+        private readonly MvcOptions _mvcOptions;
+
+        public SystemTextJsonHelper(IOptions<MvcOptions> mvcOptions)
+        {
+            _mvcOptions = mvcOptions.Value;
+        }
+
+        /// <inheritdoc />
+        public IHtmlContent Serialize(object value)
+        {
+            // JsonSerializer always encodes
+            var json = JsonSerializer.ToString(value, value.GetType(), _mvcOptions.SerializerOptions);
```

Comment:
System.Text.Json's default and only behavior is the equivalent of `StringEscapeHandling.EscapeNonAscii` (https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_StringEscapeHandling.htm). There isn't a way (as yet) to turn this behavior off.

Reply:
One thing to note is that the extension method to `IJsonHelper` that accepts a setting is currently only available when you use the Json.NET based helper: https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.NewtonsoftJson/src/JsonHelperExtensions.cs#L30-L33.

We could consider adding it in the future if somebody asks for it. As of now, the extension primarily exists to maintain parity with 2.x